src/lua/httpc.lua

```diff
@@ -44,6 +44,12 @@ local encoders = {
     ['application/msgpack'] = function(body, _content_type) return msgpack.encode(body) end,
```

Comment:
Regarding the commit message (2nd commit):

I think it is important to clarify the following nuances of the behavior:

* If a content type is not present in the response, `:decode()` assumes `application/json`.
* If a content type is present in the response, but there is no same named decoder, `:decode()` will raise an error.